internal/ingress/controller/controller_test.go

```diff
@@ -274,8 +277,9 @@ func TestCheckIngress(t *testing.T) {
 				t.Errorf("with a new catch-all ingress and catch-alls disable, should return error")
 			}
 
-			// set back to nil for next test
-			ing.Spec.Backend = nil
+			// reset backend and catch-all flag
+			ing.Spec.Backend = backendBefore
```

Comment:
Also regarding reseting these, there should be a before/after function that resets all these values. Currently the other tests are not doing this (pretty sure any new test after the `When the ingress is in a different namespace than the watched one` test will fail because namespace is not being reset). I can create a separate PR that implements this.

Reply:
Just confirmed this is true, also same thing for the `When the default annotation prefix is used despite an override` test. The only reason this test is passing is because it throws an error before the lines being tested by that test.